File: app/src/main/java/com/marekguran/tzi/MainActivity.java
package com.marekguran.tzi;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        CardView abeceda_slovo_jazyk = (CardView) findViewById(R.id.abeceda_slovo_jazyk);
        CardView konecny_automat = (CardView) findViewById(R.id.konecny_automat);
        CardView operacie_nad_jazykmi = (CardView) findViewById(R.id.operacie_nad_jazykmi);
        CardView teoria = (CardView) findViewById(R.id.teoria);

        abeceda_slovo_jazyk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, AbecedaSlovoJazyk1.class));
                finish();

            }
        });
        konecny_automat.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, KonecnyAutomat1.class));
                finish();

            }
        });
        operacie_nad_jazykmi.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, OperacieNadJazykmi1.class));
                finish();

            }
        });
        teoria.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, TeoriaMenu.class));
                finish();

            }
        });

        abeceda_slovo_jazyk.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else if (event.getAction() == MotionEvent.ACTION_UP) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
                return false;
            }
        });
        konecny_automat.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else if (event.getAction() == MotionEvent.ACTION_UP) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
                return false;
            }
        });
        operacie_nad_jazykmi.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else if (event.getAction() == MotionEvent.ACTION_UP) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
                return false;
            }
        });
        teoria.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else if (event.getAction() == MotionEvent.ACTION_UP) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
                return false;
            }
        });


        abeceda_slovo_jazyk.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View view, boolean hasFocus) {
                if (hasFocus) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
            }
        });
        konecny_automat.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View view, boolean hasFocus) {
                if (hasFocus) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
            }
        });
        operacie_nad_jazykmi.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View view, boolean hasFocus) {
                if (hasFocus) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
            }
        });
        teoria.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View view, boolean hasFocus) {
                if (hasFocus) {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                } else {
                    ObjectAnimator scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f);
                    ObjectAnimator scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f);
                    scaleDownX.setDuration(150);
                    scaleDownY.setDuration(150);

                    AnimatorSet scaleDown = new AnimatorSet();
                    scaleDown.play(scaleDownX).with(scaleDownY);

                    scaleDown.start();
                }
            }
        });
    }
}

File: app/src/main/java/com/marekguran/tzi/MainActivity.kt
package com.marekguran.tzi

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.view.View.OnFocusChangeListener
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import com.marekguran.tzi.kvizy.Kvizy
import com.marekguran.tzi.priklady.Priklady
import com.marekguran.tzi.teoria.TeoriaMenu
import com.marekguran.tzi.testy.Testy

class MainActivity : AppCompatActivity() {
    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val kvizy = findViewById<View>(R.id.kvizy) as CardView
        val priklady = findViewById<View>(R.id.priklady) as CardView
        val testy = findViewById<View>(R.id.testy) as CardView
        val teoria = findViewById<View>(R.id.teoria) as CardView
        kvizy.setOnClickListener {
            startActivity(Intent(this@MainActivity, Kvizy::class.java))
            finish()
        }
        priklady.setOnClickListener {
            startActivity(Intent(this@MainActivity, Priklady::class.java))
            finish()
        }
        testy.setOnClickListener {
            startActivity(Intent(this@MainActivity, Testy::class.java))
            finish()
        }
        teoria.setOnClickListener {
            startActivity(Intent(this@MainActivity, TeoriaMenu::class.java))
            finish()
        }
        kvizy.setOnTouchListener { view, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else if (event.action == MotionEvent.ACTION_UP) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
            false
        }
        priklady.setOnTouchListener { view, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else if (event.action == MotionEvent.ACTION_UP) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
            false
        }
        testy.setOnTouchListener { view, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else if (event.action == MotionEvent.ACTION_UP) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
            false
        }
        teoria.setOnTouchListener { view, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else if (event.action == MotionEvent.ACTION_UP) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
            false
        }
        kvizy.onFocusChangeListener = OnFocusChangeListener { view, hasFocus ->
            if (hasFocus) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
        }
        priklady.onFocusChangeListener = OnFocusChangeListener { view, hasFocus ->
            if (hasFocus) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
        }
        testy.onFocusChangeListener = OnFocusChangeListener { view, hasFocus ->
            if (hasFocus) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
        }
        teoria.onFocusChangeListener = OnFocusChangeListener { view, hasFocus ->
            if (hasFocus) {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 0.9f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 0.9f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            } else {
                val scaleDownX = ObjectAnimator.ofFloat(view, "scaleX", 1f)
                val scaleDownY = ObjectAnimator.ofFloat(view, "scaleY", 1f)
                scaleDownX.duration = 150
                scaleDownY.duration = 150
                val scaleDown = AnimatorSet()
                scaleDown.play(scaleDownX).with(scaleDownY)
                scaleDown.start()
            }
        }
    }
}